include/range/v3/view/cartesian_product.hpp

```diff
@@ -174,7 +174,7 @@ namespace ranges
                     cursor const &that, meta::size_t<1>, dist_info const inf) const
                 {
                     auto const my_distance = std::get<0>(that.its_) - std::get<0>(its_);
-                    return my_distance * inf.size_product + inf.distance;
+                    return static_cast<std::ptrdiff_t>(my_distance * inf.size_product + inf.distance);
```

Comment:
`inf.size_product` and `inf.distance` are already `ptrdiff_t`....did you get a warning for a range whose distance type is larger than `ptrdiff_t`? I can easily see this being a problem for 32-bit builds with ranges that use 64-bit difference types. The intent behind using `std::ptrdiff_t` directly - instead of something like `common_type</*...*/<Views>...>` - was to get more "clearance" for the size product.

Maybe all of the `std::ptrdiff_t` uses should be replaced with `common_type_t<std::ptrdiff_t, difference_type_t<iterator_t<Views>>...>`?

Reply:
Yes, I got a warning, and it broke Appveyor. See <https://ci.appveyor.com/project/ericniebler/range-v3/build/1.0.592>.

I would rather get this fix in and deal with the truncation issues separately. After all, that problem is not new. (I'm not sure why we're seeing it only now.) Mind if I assign you the bug? There's no rush to fix it.

Reply:
I'm fine with all of the above.